crates/emitter/src/scope_notes.rs

```diff
@@ -62,9 +62,36 @@ impl ScopeNoteList {
         ScopeNoteIndex::new(note_index)
     }
 
+    fn get_scope_hole_gcthing_index(&self, reference_scope_index: ScopeNoteIndex) -> GCThingIndex {
+        if reference_scope_index.index >= 1 {
+            return self
+                .notes
+                .get(reference_scope_index.index - 1)
+                .unwrap()
+                .index;
+        }
```

Comment:
looks like this is based on assumption that the previous item is enclosing scope.
this doesn't hold when there was an inner scope, like:

```
{ // scope note A created here
  {  // scope note B created here
    let a = 10;
  }
  for (;;) { // scope note C created here
    break; // try to refer the A, by C - 1, but it's B
  }
}
```

so, instead of this calculation, please use the emitter scope's stack to refer the enclosing scope's note index